### PR TITLE
feat(ST-44): searching, adding, removing participants on classroom added

### PR DESCRIPTION
**Pull request by tareq-mahmood-sazim**
## Description of Change

- Endpoint created for searching students by name and id
- Endpoint created for getting teachers by ID
- Endpoint created for adding students into classroom
- Endpoint created for removing students from classroom
- Authguard provided over adding and removing students to ensure role permissions.

## Please list which modules are affected by your changes

src/classrooms/*, src/users/*
---

- [x] **You assure that you have tested the modules properly and there are no breaking changes**
- [x] **You adhere to Sazim's [Engineering Principles](https://docs.google.com/document/d/10mowIwsuYMyNkXS7aeL7AJyVNMYVh1vdMf7yn-A6I1k/edit#heading=h.qqetrc8w2lm8)**

## Associated Ticket Link(s)

- https://trello.com/c/5nmO0Q4u

## Related Pull Request(s)

-

### CODE REVIEW

File: src/classrooms/classrooms.controller.ts [hunk 1]
```diff
@@ -29,6 +40,21 @@ export class ClassroomsController {
...
   }
+  @UseGuards(JwtAuthGuard)
+  @Get("participants/:id")
+  GetParticipantsByTeacherId(@Req() req: { user: TokenizedUser }, @Param("id") id: string) {
+    const teacherId = req.user.id;
+    return this.classroomsService.findAllParticipantsByTeacherId(teacherId, parseInt(id));
+  }
+  @UseGuards(JwtAuthGuard)
+  @Post("participants/remove")
+  RemoveParticipant(
```
Comment (nur-zaman-sazim) [inline src/classrooms/classrooms.controller.ts:51]: nit: please use a small r in removeParticipant

File: src/users/users.controller.ts
Comment (nur-zaman-sazim) [inline src/users/users.controller.ts:1]: Let's combine our /student/search/name/:param and /student/search/email/:param routes into a single /student/search endpoint with a query parameter (e.g., ?name=value or ?email=value).

We can even drop the /search/ part as well

File: src/users/users.controller.ts [hunk 2]
```diff
@@ -34,4 +37,26 @@ export class UsersController {
     const newUser = await this.usersService.createOne(registerUserDto);
     return this.usersSerializer.serialize(newUser);
   }
+
+  @Get("/student/:id")
+  async getStudent(@Param() param: { id: string }) {
+    const student = await this.usersService.getStudentById(parseInt(param.id));
```
Comment (nur-zaman-sazim) [inline src/users/users.controller.ts:43]: what if the ID can't be parsed?
can we handle this more gracefully?

File: src/users/users.module.ts [hunk 3]
```diff
@@ -7,13 +7,19 @@ import { RolesModule } from "@/roles/roles.module";
...
-import { UsersSerializer } from "./users.serializer";
+import { UsersSerializer, UsersProfileSerializer } from "./users.serializer";
 import { UsersService } from "./users.service";
 
 @Module({
   imports: [RolesModule, MikroOrmModule.forFeature([User])],
   controllers: [UsersController],
-  providers: [UsersService, UsersSerializer, UniquecodeRepository],
-  exports: [UsersService, UsersSerializer, MikroOrmModule.forFeature([User]), UniquecodeRepository],
+  providers: [UsersService, UsersSerializer, UniquecodeRepository, UsersProfileSerializer],
```
Comment (nur-zaman-sazim) [inline src/users/users.module.ts:16]: ooc: why do we have two different UserSerializer and userProfileSerializer

**Review by nur-zaman-sazim — CHANGES_REQUESTED** [review]

File: src/users/users.controller.ts
Comment (tareq-mahmood-sazim) [reply src/users/users.controller.ts:1]: I felt this nest was necessary as I am getting the students by id and in some cases email or name as regex. keeping single endpoint would complicate that single endpoint bhaia.

File: src/users/users.module.ts (hunk 3 above)
Comment (tareq-mahmood-sazim) [reply src/users/users.module.ts:16]: UserSerializer removes secrets from User entity, UserProfile removes secrets from User-Profile Entity

**Review by nur-zaman-sazim — APPROVED** [review]

**Pull request merged**